content/solids/components/net.ts

```diff
@@ -0,0 +1,343 @@
+// =============================================================================
+// 3D Net Component
+// (c) Mathigon
+// =============================================================================
+
+
+/// <reference types="THREE"/>
+import {register} from '@mathigon/boost';
+import {Angle, Point, Polygon, Rectangle} from '@mathigon/euclid';
+import {round} from '@mathigon/fermat';
+import {Solid} from '../../shared/components/solid';
+
+type Hinge = [number, number, number];
+
+type HingeBone = {
+  bone: THREE.Bone,
+  verts: Point[],
+  axis: THREE.Vector3,
+  maxAngle: number
+};
+
+@register('x-net')
+export class Net extends Solid {
+  private hingeBones: HingeBone[] = [];
+  private loaded!: boolean;
+
+  created() {
+    this.one('loaded', () => this.loaded = true);
+  }
+
+  private add(faces: Polygon[], hinges: Hinge[]) {
+    const netPoints = mergePolys(faces).points;
+
+    const geoInit = new THREE.Geometry();
+
+    for (const netPoint of netPoints) {
+      geoInit.vertices.push(new THREE.Vector3(netPoint.x, netPoint.y, 0));
+    }
+
+    const facesTris = faces.map(triangulateFace);
+
+    for (const faceTris of facesTris) {
+      for (const tri of faceTris) {
+        const [a, b, c] = tri.map(point => netPoints.findIndex(p => p.equals(point)));
+        geoInit.faces.push(new THREE.Face3(a, b, c));
+      }
+    }
+
+    const baseCenter = faces[0].centroid;
+
+    const rootBone = new THREE.Bone();
+    rootBone.name = 'root';
+    rootBone.position.set(baseCenter.x, baseCenter.y, 0);
+    const rootHingeBone: HingeBone = {
+      bone: rootBone,
+      verts: faces[0].points,
+      axis: new THREE.Vector3(0, 0, 0),
+      maxAngle: 0
+    };
+
+    const hingeBonesTail =
+      boneHingesForFace(
+          0,
+          null,
+          rootBone,
+          faces,
+          hinges
+      );
+    const hingeBones = [rootHingeBone, ...hingeBonesTail];
+    this.hingeBones = hingeBones;
+
+    const geometry =
+      new THREE.BufferGeometry().fromGeometry(geoInit);
+
+    const skinIndices: number[] = [];
+    const skinWeights: number[] = [];
+
+    const position = geometry.attributes.position as THREE.BufferAttribute;
+    const vertex = new THREE.Vector3();
+    for (let i = 0; i < position.count; i++) {
+      vertex.fromBufferAttribute(position, i);
+
+      const testPoint = new Point(vertex.x, vertex.y);
+      const boneIndex = getBoneIndexFromPoint(testPoint, hingeBones);
+
+      skinIndices.push(boneIndex, 0, 0, 0);
+      skinWeights.push(1, 0, 0, 0);
+    }
+
+    // TODO: Change to .setAttribute if threejs version is updated
+    geometry.addAttribute('skinIndex', new THREE.Uint16BufferAttribute(skinIndices, 4));
+    geometry.addAttribute('skinWeight', new THREE.Float32BufferAttribute(skinWeights, 4));
+
+    const mat = new THREE.MeshPhongMaterial({
```

Comment:
You could use this existing method for materials: https://github.com/mathigon/textbooks/blob/master/content/shared/components/solid.ts#L328

And add one more attribute for `skinning` that defaults to false? Just to make sure that all the materials are consistent